src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -1504,10 +1504,13 @@ def generate_snapshot(self, rom_name: str, on_disk: bool = True, named_selection
             msg += f"\n{str(e)}."
             self._raise_error(msg)
 
-    def generate_snapshot_batch(self, batch_results: pd.DataFrame, rom_name: str, named_selection: str = None):
+    def generate_snapshot_batch(
+        self, batch_results: pd.DataFrame, rom_name: str, on_disk: bool = True, named_selection: str = None
+    ):
         """
-        Generate several field snapshots based on historical batch results of the Twin, for the full field or a specific
-        named selection. It returns a list of the paths of the different snapshots written on disk.
+        Generate several field snapshots based on historical batch results of the Twin, either in memory or on disk, for
+        the full field or a specific named selection. It returns a list of the field data as an array if in memory, or a
+        list of the paths of the different snapshots written on disk.
```

Comment:
Possibly add a Returns section to make this clearer e.g.
```
Returns
--------
list[str] | list[np.ndarray]
    list of paths to snapshots written to disk if on_disk == True, else list of arrays of snapshot field data
```

Reply:
I have included these changes and adopted them for other similar methods, thanks